src/cascadia/TerminalSettingsModel/TerminalSettings.cpp

```diff
@@ -88,53 +96,100 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+                settings.DefaultSettings().SuppressApplicationTitle(newTerminalArgs.SuppressApplicationTitle().Value());
             }
         }
 
         return settings;
     }
 
+    void TerminalSettings::_ApplyAppearanceSettings(const IAppearanceConfig& appearance, const Windows::Foundation::Collections::IMapView<winrt::hstring, ColorScheme>& schemes)
+    {
+        _CursorShape = appearance.CursorShape();
+        _CursorHeight = appearance.CursorHeight();
+        if (!appearance.ColorSchemeName().empty())
+        {
+            if (const auto scheme = schemes.TryLookup(appearance.ColorSchemeName()))
+            {
+                ApplyColorScheme(scheme);
+            }
+        }
+        if (appearance.Foreground())
+        {
+            _DefaultForeground = til::color{ appearance.Foreground().Value() };
+        }
+        if (appearance.Background())
+        {
+            _DefaultBackground = til::color{ appearance.Background().Value() };
+        }
+        if (appearance.SelectionBackground())
+        {
+            _SelectionBackground = til::color{ appearance.SelectionBackground().Value() };
+        }
+        if (appearance.CursorColor())
+        {
+            _CursorColor = til::color{ appearance.CursorColor().Value() };
+        }
+        if (!appearance.BackgroundImagePath().empty())
+        {
+            _BackgroundImage = appearance.ExpandedBackgroundImagePath();
+        }
+
+        _BackgroundImageOpacity = appearance.BackgroundImageOpacity();
+        _BackgroundImageStretchMode = appearance.BackgroundImageStretchMode();
+        std::tie(_BackgroundImageHorizontalAlignment, _BackgroundImageVerticalAlignment) = ConvertConvergedAlignment(appearance.BackgroundImageAlignment());
+
+        _RetroTerminalEffect = appearance.RetroTerminalEffect();
+        _PixelShaderPath = winrt::hstring{ wil::ExpandEnvironmentStringsW<std::wstring>(appearance.PixelShaderPath().c_str()) };
+    }
+
     // Method Description:
     // - Creates a TerminalSettings object that inherits from a parent TerminalSettings
     // Arguments::
     // - parent: the TerminalSettings object that the newly created TerminalSettings will inherit from
     // Return Value:
     // - a newly created child of the given parent object
-    Model::TerminalSettings TerminalSettings::CreateWithParent(const Model::TerminalSettings& parent)
+    Model::TerminalSettingsStruct TerminalSettings::CreateWithParent(const Model::TerminalSettingsStruct& parent)
     {
         THROW_HR_IF_NULL(E_INVALIDARG, parent);
 
-        auto parentImpl{ get_self<TerminalSettings>(parent) };
-        return *parentImpl->CreateChild();
+        auto defaultImpl{ get_self<TerminalSettings>(parent.DefaultSettings()) };
+        auto defaultChild = defaultImpl->CreateChild();
+        if (parent.UnfocusedSettings())
+        {
+            parent.UnfocusedSettings().SetParent(*defaultChild);

```

Comment:
wait wait wait, you're editing the _parent's_ parent. That seems like it is VERY BAD

Reply:
Replied above! Resolving this so the discussion can be in one place

Reply:
I legitimately do not think this is correct. We're saying "Give me a new settings object that derives from this parent", and then the parent is edited so that it is the CHILD.

## BEFORE

```
┌──────────────────────┐
│                      │
│ Original Settings    │
│                      │
│                      │
│                      │
│                      │
├──────────────────────┘
│
│ ┌────────────────────┐
│►│ Default Appearance │
│ │                    │
│ └──────────────────▲─┘
│                    │
│ ┌──────────────────┴─┐
└►│ Unfocused Appear.  │
  │                    │
  └────────────────────┘
```

## AFTER

```


    ┌──────────────────────┐        ┌─────────────────────────┐
    │                      │        │                         │
    │ Original Settings    │        │    NEW CHILD SETTINGS   │
    │                      │        │                         ├──┐
    │                      │        │                         │  │
    │                      │        │                         │  │
    │                      │        │                         │  │
    ├──────────────────────┘        └─────────────────────────┘  │
    │                                                            │
    │ ┌────────────────────┐       ┌────►┌────────────────────┐  │
    │►│ Default Appearance │       │     │ Default Appearance │◄─┘
    │ │                    │◄──────┼─────┤                    │
    │ └───────────────────┴┘       │     └────────────────────┘
    │                              │     
    │ ┌────────────────────┐       │     
    └►│ Unfocused Appear.  ├───────┘     
      │                    │ this part   
      └────────────────────┘ i do not    
                             understand

```

Reply:
Doesn't this mean that somebody who was using the old UnfocusedSettings now gets different settings through it?

Am I just thinking of inheritance incorrectly, and this needs to be very well documented?

Reply:
(I see you added a comment, but the comment explains what it does, not why it does it!)

Reply:
Added more comments explaining this!

Reply:
Discussed offline. Yeah, this makes sense. I guess what was confusing to me is that this function would make `TerminalSettings` inherit from another `TerminalSettings`. Here, we're doing the same thing. `TerminalSettingsCreateResult` is intended to just be a way to pass two objects at once. It is _not_ inheritable, whereas `TerminalSettings` is.